### ISSUE TITLE
Bug: Search form issue - Search should be case-insensitive

### ISSUE THREAD

**Issue by audserraCGI** [CGI-FR/IoT-Hub-Portal#567]
## Expected Behavior
When using the search form to filter the device list, searching by name/ID should be case-insensitive.

## Current Behavior
Searching "device" doesn't allow us to retrieve our mock device called "Device 1"

## Additional Information

There's also a pragma warning in the code asking to normalize the filtering:
https://github.com/CGI-FR/IoT-Hub-Portal/blob/5541bb326d1c01cb3f9b7fdfb723070a70adde93/src/AzureIoTHub.Portal/Server/Services/DeviceService.cs#L59-L61

**Comment by kbeaugrand** [comment]
Currently, IoT Hub query language doesn't provide functions that can be used to query easily the devices (with lowercase, starts_with, contains, ...).

We decided to postpone this implementation until these features are available in IoT Hub query language. 

Se [https://docs.microsoft.com/en-US/azure/iot-hub/iot-hub-devguide-query-language](https://docs.microsoft.com/en-US/azure/iot-hub/iot-hub-devguide-query-language)

**Comment by kbeaugrand** [comment]
Blocked by #1029, #1030, and #1031

**Comment by hocinehacherouf** [comment]
Fix by #862
